meilisearch/_httprequests.py

```diff
@@ -94,6 +95,14 @@ def delete(
     ) -> Any:
         return self.send_request(requests.delete, path, body)
 
+    @lru_cache(maxsize=1)
+    def _build_user_agent(self, client_agents: Optional[Tuple[str]] = None) -> str:
+        user_agent = qualified_version()
+        if not client_agents:
+            return user_agent
+
+        return f"{user_agent};{';'.join(client_agents)}"
+
```

Comment:
You will want to move this out of the class or it can lead to memory leaks. If you want me to push an update so you can see what I mean let me know.